packages/openapi-typescript/src/transform/path-item-object.ts

```diff
@@ -104,11 +104,11 @@ export default function transformPathItemObject(
     }
     // if operationId exists, move into an `operations` export and pass the reference in here
     else if (operationObject.operationId) {
-      operationType = oapiRef(
-        createRef(["operations", operationObject.operationId]),
-      );
+      // workaround for issue caused by redocly ref parsing: https://github.com/drwpow/openapi-typescript/issues/1542
+      const operationId = operationObject.operationId.replace(/#/g, "/");
```

Comment:
For RegExs, it’s good to pull them out into the top-level module scope as constants, that way they don’t recompile every function run. This one may be negligible, but some RegExs can perceptibly slow things down especially when looping; better to just do it regardless.

```suggestion
      const operationId = operationObject.operationId.replace(HASH_RE, "/");
```

Reply:
Just confirming (to myself) I’m not crazy and this is still valid: still a perceptible difference in V8 between the two:

![CleanShot 2024-02-15 at 20 48 34@2x](https://github.com/drwpow/openapi-typescript/assets/1369770/393d1c41-9abb-47bc-ba0b-5e04f2449527)

17% isn’t a lot, but this is the best-case scenario. Some larger RegExs can be orders of magnitude slower. Also, in application-level code this is splitting hairs, but for libraries like this I think splitting hairs like this can be helpful to the end user.

Reply:
For sure- good call